src/LiveComponent/src/EventListener/AddLiveAttributesSubscriber.php

```diff
@@ -75,7 +76,7 @@ private function getLiveAttributes(MountedComponent $mounted): ComponentAttribut
             'data-live-data-value' => twig_escape_filter($twig, json_encode($data, \JSON_THROW_ON_ERROR), 'html_attr'),
         ];
 
-        if ($this->container->has(CsrfTokenManagerInterface::class)) {
+        if ($this->container->has(CsrfTokenManagerInterface::class) && (new \ReflectionObject($mounted->getComponent()))->getAttributes(AsLiveComponent::class)[0]->newInstance()->csrf) {
```

Comment:
You should make this available as "metadata" on the mounted component.

Add the `AsLiveComponent::$csrf` parameter to the array [here](https://github.com/symfony/ux/blob/e4ab138168b481c051cb63b80b6f20d19c5d03c3/src/LiveComponent/src/Attribute/AsLiveComponent.php#L39). 

```suggestion
        if ($this->container->has(CsrfTokenManagerInterface::class) && $metadata->get('csrf')) {
```

Reply:
I thought about that too, but didn't understand the metadata system, so did another way. :)

Reply:
This will also technically allow this config to be set by _container config only_ (the attribute isn't required).